### DISCUSSION TITLE
build: windows image

### DISCUSSION THREAD

**Discussion by jjangga0214** [verdaccio/verdaccio#4867] [Ideas]
Hi! 

Thanks for the valuable project :) 

I am running verdaccio in GitHub Actions for Linux, macOS, Windows. 
To run Linux-based container images on Windows, `Hiper-V` should be turned on.
But it's turned off on GitHub Actions' Windows runners and users cannot change this setting.
Thus we cannot run verdaccio from Docker on Windows.

As a workaround, I am running it naively through `npm`, which is not ideal in my case.
Therefore I hope Windows-based images to be provided.

Thanks :)

**Comment by mbtools** [comment]
I'm running [Docker Desktop](https://docs.docker.com/desktop/install/windows-install/) on Win 10 + WSL2. The [verdaccio image](https://hub.docker.com/r/verdaccio/verdaccio) is running fine with that.

![image](https://github.com/user-attachments/assets/7c255eda-cc60-4ee1-9a94-0e717aea85ee)

**Comment by jjangga0214** [comment]
@mbtools AFAIK, wsl2 can not be enabled for docker due to disabled Hiper-V in GitHub Actions.

**Comment by mbtools** [comment]
As you can see in my screenshot, it works. Why do you need GH actions? 

It all runs on my Win10 PC with Docker Desktop. You can use the Verdaccio images from Docker Hub or clone the repo to your PC and run docker-compose.

**Comment by jjangga0214** [comment]
@mbtools Thanks for the response. 

> As you can see in my screenshot, it works.

I know, but that's thanks to wsl2.
You are running `linux/amd64` image on Windows by wsl2.

> Why do you need GH actions?

I maintain an os-sensitive package that has `postinstall` npm hook.
So on CI, which is GitHub Actions in my case, I want an e2e-test on Windows, Linux, and macOS.
I want to run local verdaccio on each OS, and test publishing and installation.
I can install verdaccio by npm, but I preferred a docker image for some reasons.

> It all runs on my Win10 PC with Docker Desktop. You can use the Verdaccio images from Docker Hub or clone the repo to your PC and run docker-compose.

I know all about that. That's not what this issue is for.

**Comment by mbtools** [comment]
Got it. The image needs to run on GH actions. 

It looks to me like creating such windows images is still experimental and a manual process:

https://docs.docker.com/build/buildkit/#buildkit-on-windows

I'm not sure how long it will take until this becomes an option for automating it here:

https://github.com/verdaccio/verdaccio/blob/695c28b5f26fe56b242f02dff178ba6d210b8953/.github/workflows/docker-publish.yml#L54
